### PR TITLE
you can now use any typepath as an emoji :/obj/item/clothing/mask/joy:

### PR DESCRIPTION
**Pull request by silicons**
## About The Pull Request

title

## Why It's Good For The Game

more emojis.
this won't look good for a LOT of items because overlay fuckery but uh, hey, it works!

## Changelog
:cl: silicons
add: You can now use anything as an emoji by doing :/obj/item/path/to/item:. This works for any /atom or subtype.
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
:pickle:

**Comment by LetterN** [comment]
make sure to blacklist some paths!

File: code/modules/emoji/emoji_parse.dm [hunk 1]
```diff
@@ -16,11 +16,16 @@
...
 				emoji = lowertext(copytext(text, pos + length(text[pos]), search))
+				var/isthisapath = (emoji[1] == "/") && text2path(emoji)
 				var/datum/asset/spritesheet/sheet = get_asset_datum(/datum/asset/spritesheet/goonchat)
 				var/tag = sheet.icon_tag("emoji-[emoji]")
 				if(tag)
 					parsed += "<i style='width:16px !important;height:16px !important;'>[tag]</i>" //evil way of enforcing 16x16
 					pos = search + length(text[pos])
+				else if(ispath(isthisapath, /atom))	//path
+					var/atom/thisisanatom = isthisapath
+					parsed += "[icon2html(initial(thisisanatom.icon), world, initial(thisisanatom.icon_state))]"
```
Comment (LetterN) [inline code/modules/emoji/emoji_parse.dm:27]: GLOB.clients instead of world i think

Comment (silicons) [reply code/modules/emoji/emoji_parse.dm:27]: does it do anything different? think the proc just goes to glob.clients right?

**Comment by silicons** [comment]
debugging

**Comment by silicons** [comment]
this works now

**Comment by silicons** [comment]
issues with stuff not displaying

**Comment by Ludox235** [comment]
This may or may not fuck up the actual normal emojis and the RPDs.

**Comment by silicons** [comment]
seems to be fine now.

**Comment by Ghommie** [comment]
I'm a bit concerned about sprites bigger than 32 x 32, especially singularities and fullscreen hud stuff.

**Comment by ghost** [comment]
Regarding the icon size, goonchat resizes them to 16x16, and for some reason the emoji code also resizes them before that.

**Comment by Ghommie** [comment]
Good.

**Pull request merged**